java/src/jmri/jmrix/SystemConnectionMemo.java

```diff
@@ -147,11 +147,11 @@ public final boolean setUserName(@Nonnull String userName) {
      * @return true if the specified manager is provided
      */
     public boolean provides(Class<?> c) {
-        return false; // nothing, by default
+        return (get(c) != null);
```

Comment:
The provides method may be used to check to see if a system CAN provide a system of a given type.  It may actually be called before the objects are initialized, so the get method MAY return null during initialization for an object the system can actually provide.

Reply:
Which would be 100% ok… considering that this method would have returned false anyway.

Reply:
Most sub-classes override this method and test (c instanceof XxxManager)… Just seemed like overkill to duplicate all that code in the get method…

Reply:
That may be, but that's the pattern we use.  It's a bit of explicit documentation inside the memo.

Reply:
Ok, I'm not married to it… Probably means that my sub-class will have to implement it thou.